File: easy/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 1
        max = 0
        while r <= len(prices)-1:
            if prices[l] > prices[r]:
                l += 1
                r = l+1
            else:
                if prices[r] - prices[l] > max:
                    max = prices[r] - prices[l]
                r = r + 1
        return max

File: easy/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int res=0;
        int msf=Integer.MAX_VALUE;
        for (int i=0; i<prices.length; ++i) {
            res=Math.max(res,prices[i]-msf);
            msf=Math.min(msf,prices[i]);
        }
        return res;
    }
}
